examples/key-value-store/src/net_ffi.cc

```diff
@@ -0,0 +1,55 @@
+#include "net_ffi.hh"
+
+namespace net_ffi {
+
+std::unique_ptr<server_socket> listen(uint16_t port) {
+    seastar::socket_address address(0, port);
+
+    seastar::listen_options options;
+    options.proto = seastar::transport::TCP;
+    options.reuse_address = true;
+
+    server_socket socket = seastar::listen(address, options);
+    return std::make_unique<server_socket>(std::move(socket));
+}
+
+VoidFuture accept(
+    const std::unique_ptr<server_socket>& server_socket,
+    std::unique_ptr<connected_socket>& socket
+) {
+    seastar::accept_result result = co_await server_socket->accept();
+    socket = std::make_unique<connected_socket>(std::move(result.connection));
+}
+
+std::unique_ptr<input_stream> get_input_stream(
+    const std::unique_ptr<connected_socket>& socket
+) {
+    input_stream input = socket->input();
+    return std::make_unique<input_stream>(std::move(input));
+}
+
+std::unique_ptr<output_stream> get_output_stream(
+    const std::unique_ptr<connected_socket>& socket
+) {
+    output_stream output = socket->output();
+    return std::make_unique<output_stream>(std::move(output));
+}
+
+VoidFuture close_output_stream(const std::unique_ptr<output_stream>& output) {
+    co_await output->close();
+}
+
+StringFuture read(const std::unique_ptr<input_stream>& input) {
+    auto buffer = co_await input->read();
+    co_return rust::String(buffer.begin(), buffer.size());
+}
+
+VoidFuture write(
+    const std::unique_ptr<output_stream>& output,
+    const rust::str msg
+) {
+    co_await output->write(seastar::sstring(msg.begin(), msg.size()));
+    co_await output->flush();
+}
```

Comment:
Flush should be a separate method.

Reply:
I made a new function `net_ffi:flush_output`. This change not only makes sense stylistically but also enables a tiny improvement. There is no reason to flush the output stream after every write. One message from a client can contain many requests. Flushing after responding to all of them is sufficient. That is why the new `net_ffi:flush_output` is called after responding to all requests in the `ShardedServer::handle_connection` method.

Reply:
What if a client send a request, waits for response and only then sends another one? Isn't flush after each request helpful in this scenario?

Reply:
In such a case, everything would work fine. The server would instantly flush the output stream after each request. The difference after the change is that the server flushes after processing a single message (a single `read`) and not after processing a single request. Usually, there is one request per message, but the client can send multiple requests in a single message. Then, this change matters.

I still think it always works fine, but maybe I am wrong, and there is some much weirder scenario when this change is detrimental. It also makes the code less understandable. I think code that for sure works and is easier to understand is much better in a demo, so I decided to undo this change.